io_scene_vox.py

```diff
@@ -203,53 +220,147 @@ def import_vox(path, *, voxel_spacing=1, voxel_size=1,
     base_voxel = bpy.context.object
     if use_palette:
         base_voxel.active_material = mat_palette[voxel[3]]
 
+    print("Loaded voxels:", len(voxels))
+
     to_link = []
 
     for voxel in voxels:
         copy = base_voxel.copy()
         copy.data = base_voxel.data.copy()
         copy.location = [float(coord) * voxel_spacing for coord in voxel[:3]]
+        #copy.parent = base_voxel
         to_link.append(copy)
         if use_palette:
             copy.active_material = mat_palette[voxel[3]]
 
     for object_ in to_link:
-        bpy.context.scene.objects.link(object_)
+        bpy.context.scene.collection.objects.link(object_)
+
+    print("Linked voxels:", len(to_link))
 
-    bpy.context.scene.update()
+    layer = bpy.context.view_layer
+    layer.update()
 
     print('\nSuccessfully imported {} in {:.3f} sec'.format(path, time.time() - time_start))
     return {'FINISHED'}
 
 
+#
+#==================================================================================================
+#     Change Diffuse shader to Emission shader without affecting shader color
+#     http://web.purplefrog.com/~thoth/blender/python-cookbook/change-diffuse-to-emission-node.html 
+#==================================================================================================
+#
+import bpy
+
+def replace_with_emission(node, node_tree):
+    new_node = node_tree.nodes.new('ShaderNodeEmission')
+    connected_sockets_out = []
+    sock = node.inputs[0]
+    if len(sock.links)>0:
+        color_link = sock.links[0].from_socket
+    else:
+        color_link=None
+    defaults_in = sock.default_value[:]
+
+    for sock in node.outputs:
+        if len(sock.links)>0:
+            connected_sockets_out.append( sock.links[0].to_socket)
+        else:
+            connected_sockets_out.append(None)
+
+    #print( defaults_in )
+
+    new_node.location = (node.location.x, node.location.y)
+
+    if color_link is not None:
+        node_tree.links.new(new_node.inputs[0], color_link)
+    new_node.inputs[0].default_value = defaults_in
+
+    if connected_sockets_out[0] is not None:
+        node_tree.links.new(connected_sockets_out[0], new_node.outputs[0])
+
+
+def material_diffuse_to_emission(mat):
+
+    doomed=[]
+    for node in mat.node_tree.nodes:
+        if node.type=='BSDF_DIFFUSE' or node.type=='BSDF_PRINCIPLED':
+            replace_with_emission(node, mat.node_tree)
+            doomed.append(node)
+        else:
+            print(node.type)
+
+    # wait until we are done iterating and adding before we start wrecking things
+    for node in doomed:
+        mat.node_tree.nodes.remove(node)
+
+
+def replace_on_selected_objects():
+    mats = set()
+    for obj in bpy.context.scene.objects:
+        if obj.select_set:
+            for slot in obj.material_slots:
+                mats.add(slot.material)
+
+    for mat in mats:
+        material_diffuse_to_emission(mat)
+
+def replace_in_all_materials():
```

Comment:
Also never called, presumably came wholesale with the shadeless solution

Reply:
Yeah, exactly. I just imported the whole code and functions.